### PR TITLE
Refactor add windows

### PR DESCRIPTION
**Pull request by elnica6**
here's a list of what is included in this PR:

* remove deprecated code
* refactor existing linux (el & debian) to leverage native Chef package resource as that is built into the client. No need to have a unique package resource
* add ability to forgo IP Tables configuration by providing a nil value for default['alertlogic']['agent']['firewall'] attribute in the event your system is not running IP tables
* added a default.rb recipe which handles node.platform_family support. Now you can simply add the cookbook to your runlist and the default.rb recipe will figure out what OS and platform family you are running and whether you are supported or not
* updated chef_solo.sh and cloud-init examples to run the default.rb recipe

This has been tested on Windows server 2012 R2, Amazon Linux v2014.09.02, and Ubuntu 14.04

### CODE REVIEW

**Comment by ddosia** [comment]
Please split this PR accourding to changes it brings: if you depricate something - make it a PR; if you add new functionality - send PR; if you doing refactoring - send PR.

File: attributes/default.rb [hunk 1]
```diff
@@ -2,8 +2,9 @@
 ## log-agent
 #
 default["alertlogic"]["agent"]["pkg_base_url"] = "https://scc.alertlogic.net/software"
-default["alertlogic"]["agent"]["pkg_vsn"]["deb"] = "_LATEST_"
-default["alertlogic"]["agent"]["pkg_vsn"]["rpm"] = "-LATEST-1."
+default["alertlogic"]["agent"]["pkg_vsn"]["debian"] = "_LATEST_"
```
Comment (ddosia) [inline attributes/default.rb]: why did you change key names?

Comment (elnica6) [inline attributes/default.rb]: I changed the distro key names so that they match the package extension. We dynamically deduce the package extension based on the Ohai attribute node[:platform_family] in the default.rb recipe

Comment (ddosia) [inline attributes/default.rb]: This is probably undesirable as soon as this will introduce compatibility issues with existing customers.

Comment (elnica6) [inline attributes/default.rb]: as long as customers are pinning to a specific version(which I hope they are doing), this should not cause issues. changing the attribute to the native ohai platform_family sytnax also allows us to dynamically build the url in in linux_agent.rb recipe line 69

Comment (ddosia) [inline attributes/default.rb]: I am sorry, I don't see enougth reasons to bring this incompatibility.

**Comment by motobob** [comment]
@elnica6 please revert the key names mentioned by @ddosia and PR will be merged right after. That will probably require one more key map on your side. 
Going forward, please separate the PRs into smaller chunks by functionality. That helps a lot.

**Comment by elnica6** [comment]
@motobob @ddosia I've reverted the key names to maintain backwards compatibility as you have requested. This has all been tested on ubuntu 14.04 LTS, Amazon Linux 2015.03.00 (RHEL v6), and Windows server 2012 R2. Please take a look and let me know what you think.

Thanks for your patience :)

**Comment by motobob** [comment]
@elnica6 thank you for hard work.
